api/app/Console/Commands/CreateDigitalCareersPoolsCandidates.php

```diff
@@ -0,0 +1,219 @@
+                                                                    ->toArray();
+
+            if (in_array($classificationId, $candidateClassificationsArray)) {
+                return true;
+            }
+            if ($expectedSalary) {
+                $minSalaryInt = intval($classificationMinSalary);
+                $maxSalaryInt = intval($classificationMaxSalary);
+                $salaryArray = [];
+                // $expectedSalary is an array of strings, need to convert to array of array pairs, each pair creating lower/upper limits of ints
+                foreach ($expectedSalary as $salary) {
+                    switch($salary) {
+                        case '_50_59K':
+                            array_push($salaryArray, [50000, 59999]);
+                            break;
+                        case '_60_69K':
+                            array_push($salaryArray, [60000, 69999]);
+                            break;
+                        case '_70_79K':
+                            array_push($salaryArray, [70000, 79999]);
+                            break;
+                        case '_80_89K':
+                            array_push($salaryArray, [80000, 89999]);
+                            break;
+                        case '_90_99K':
+                            array_push($salaryArray, [90000, 99999]);
+                            break;
+                        case '_100K_PLUS':
+                            array_push($salaryArray, [100000, 999999]);
+                            break;
+                    }
+                }
+
+                // having created the array of arrays with paired ints, step through each one and see if a classification's salary range lines up
+                // a match is when a salary int is MORE than the lower limit AND LESS than the upper limit
+                foreach ($salaryArray as $salaryPair) {
+                    if ($salaryPair[0] <= $minSalaryInt && $salaryPair[1] >= $minSalaryInt) {
+                        return true;
+                    }
+                    if ($salaryPair[0] <= $maxSalaryInt && $salaryPair[1] >= $maxSalaryInt) {
+                        return true;
+                    }
+                }
+            }
+            return false;
+        }
+
+        /**
+         * check whether a candidate can match a pool stream
+         * @return boolean
+        */
+        function candidateMatchesStream(array $userWithAssets, string $stream): bool {
+            global $assetArrayMap, $assetKeyToStreamMap;
+
+            $assetsArray = $userWithAssets['cmo_assets']; // grab array of assets models from the passed in collection
+            // convert array of user assets models to array of asset id's
+            $assetsIdArray = [];
+            foreach ($assetsArray as $asset) {
+                array_push($assetsIdArray, $asset['pivot']['cmo_asset_id']);
+            }
```

Comment:
This is working, for sure, but I recommend passing in the same `$userModelForCandidate` that we pass to `candidateMatchesClassification`, and getting the asset ids via the Eloquent relation and the very handy [pluck method](https://laravel.com/docs/9.x/collections#method-pluck) on Eloquent collections. I find it more succinct and more legible:
```suggestion
        function candidateMatchesStream(object $userModelForCandidate, string $stream): bool {
            global $assetArrayMap, $assetKeyToStreamMap;

            // get ids of all CmoAssets associated with the user
            $assetsIdArray = $userModelForCandidate->cmoAssets->pluck('id');
```